README.md

```diff
@@ -532,6 +538,251 @@ Following commands able you to get some additional information about token:
 ./bin/neo-go contract invokefunction -e http://localhost:20331 -w my_wallet.json -g 0.00001 f84d6a337fbc3d3a201d41da99e86b479e7a2554 totalSupply
 ```
 
+## Workshop. Part 3
+In this part we'll summarise our knowledge about smart contracts by investigating [4-domain](https://github.com/nspcc-dev/neo-go-sc-wrkshp/blob/master/4-domain.go) smart contract. This contract 
+contains code for domain registration, transferring, deletion and getting information about registered domains.
+
+Let’s go!
+
+#### Step #1
+Let's take a glance at our [contract](https://github.com/nspcc-dev/neo-go-sc-wrkshp/blob/master/4-domain.go) and inspect it. The contract takes an action string as the first parameter, which is one of the following:
+- `register` checks, whether domain with the specified name already exists. If not, it also adds the pair `[domain_name, owner]` to the storage. It requires additional arguments:
+   - `domain_name` which is the new domain name.
+   - `owner` - the 34-digit account address from our [wallet](https://github.com/nspcc-dev/neo-go-sc-wrkshp/blob/master/my_wallet.json), which will be used for contract invocation.
+- `query` returns the specified domain owner address (or false, if no such domain was registered). It requires the following argument:
+   - `domain_name` which is requested domain name.
+- `transfer` transfers domain with the specified name to the other address (of course, in case if you're the actual owner of the domain requested). It requires additional arguments:
+   - `domain_name` which is the name of domain you'd like to transfer.
+   - `to_address` - the account address you'd like to transfer the specified domain to.
+- `delete` deletes specified domain from the storage. The arguments:
+   - `domain_name` which is the name of the domain you'd like to delete.
+ 
+ In the next steps we'll compile and deploy smart contract. 
+ After that we'll try to register new domain, transfer it to another account and query information about it.
+
+#### Step #2
+
+Compile smart contract [4-domain.go](https://github.com/nspcc-dev/neo-go-sc-wrkshp/blob/master/4-domain.go)
+```
+./bin/neo-go contract compile -i 4-domain.go
+```
+
+... and deploy it with [configuration](https://github.com/nspcc-dev/neo-go-sc-wrkshp/blob/master/4-domain.yml):
+```
+./bin/neo-go contract deploy -i 4-domain.avm -c 4-domain.yml -e http://localhost:20331 -w my_wallet.json -g 100
```

Comment:
`-g 100` is certainly not needed these days, you either can omit it or just add some gas for higher priority (`-g 0.001`).

Reply:
I was wondering, why I'm running out of GAS so fast :D 
Fixed.